lib/stanford-mods/name.rb

```diff
@@ -53,10 +53,13 @@ def additional_authors_w_dates
       def non_collector_person_authors
         result = []
         @mods_ng_xml.personal_name.map do |n|
-          r = n.role
-          unless (r.authority.include?('marcrelator') && r.value.include?('Collector')) ||
-                  r.roleTerm.valueURI.first == COLLECTOR_ROLE_URI
-            result << n.display_value_w_date
+          unless n.role.size == 0
+            n.role.each { |r|
+              unless (r.authority.include?('marcrelator') && r.value.include?('Collector')) ||
+                      r.roleTerm.valueURI.first == COLLECTOR_ROLE_URI
```

Comment:
It seems like this is a near duplicate of the code below.  Perhaps we should put this into a private method and DRY this logic up a bit?

That also could let us clean up the `each` into something like this here:
```ruby
result << n.role.reject { |r| marc_relator_includes_collection_role(r) }
```
and this below:
```ruby
result << n.role.select { |r| marc_relator_includes_collection_role(r) }
```
...or whatever makes sense.

Just a thought